src/proto/beats.proto

```diff
@@ -10,22 +11,32 @@ enum Source {
   USER = 2;
 }
 
-message Beat {
-  optional int32 frame_position = 1;
-  optional bool enabled = 2 [ default = true ];
-  optional Source source = 3 [ default = ANALYZER ];
+enum Type {
+  BEAT = 0;
+  BAR = 1;
+  PHRASE = 2;
+}
+
+message TimeSignature {
+  optional int32 beats_per_bar = 1 [ default = 4];
+  optional int32 note_value = 2 [ default = 4 ];
 }
 
 message Bpm {
   optional double bpm = 1;
   optional Source source = 2 [ default = ANALYZER ];
 }
 
-message BeatMap {
-  repeated Beat beat = 1;
+message Beat {
+  // The frame position should always be explicitly set
+  // otherwise serialization would lead to SIGABRT.
+  optional double frame_position = 1 [ default = 0 ];
+  optional bool enabled = 2 [ default = true ];
+  optional Source source = 3 [ default = ANALYZER ];
+  optional Type type = 4 [ default = BEAT ];
+  optional TimeSignature signature = 5;
 }
 
-message BeatGrid {
-  optional Bpm bpm = 1;
-  optional Beat first_beat = 2;
+message Beats {
+  repeated Beat beat = 1;
 }
```

Comment:
Did we make progress here? I have strong concerns that this proposed proto buff will not last for long. 

I am not insisting to switch to the one I have proposed, but at least we should describe a our workflow , check how we will use it and verify that it fits. 

This is IMMO a strong requirement before this is merged and we start altering users data.